palo_alto_pan_os/komand_palo_alto_pan_os/util/pan_os_requests.py

```diff
@@ -182,68 +105,68 @@ def show_(self, xpath: str) -> dict:
-
-        if output['response']['@status'] == 'error':
-            error = output['response']['msg']
-            error = json.dumps(error)
-            raise PluginException(cause='PAN-OS returned an error in response to the request.',
-                                  assistance='Double that check inputs are valid. Contact support if this issue persists.',
-                                  data=error)
+        response = self.make_request("GET", querystring)
+        output = self.get_output_with_exceptions(response)
         return output
 
     def op(self, cmd: str) -> dict:
         querystring = {"type": "op", "key": self.key, "cmd": cmd}
 
-        response = self.session.get(self.url, params=querystring, verify=self.verify_cert)
-        try:
-            output = xmltodict.parse(response.text)
-        except TypeError:
-            raise PluginException(cause='The response from PAN-OS was not the correct data type.',
-                                  assistance='Contact support for help.',
-                                  data=response.text)
-        except SyntaxError:
-            raise PluginException(cause='The response from PAN-OS was malformed.',
-                                  assistance='Contact support for help.',
-                                  data=response.text)
-        except BaseException as e:
-            raise PluginException(cause='An unknown error occurred when parsing the PAN-OS response.',
-                                  assistance='Contact support for help.',
-                                  data=f'{response.text}, error {e}')
-
-        if output['response']['@status'] == 'error':
-            error = output['response']['msg']
-            error = json.dumps(error)
-            raise PluginException(cause='PAN-OS returned an error in response to the request.',
-                                  assistance='Double that check inputs are valid. Contact support if this issue persists.',
-                                  data=error)
+        response = self.make_request("GET", querystring)
+        output = self.get_output_with_exceptions(response)
         return output
 
     def commit(self, action: str, cmd: str) -> dict:
         querystring = {"type": "commit", "action": action,
                        "key": self.key, "cmd": cmd}
 
-        response = requests.get(self.url, params=querystring, verify=self.verify_cert)
+        response = self.make_request("REQUEST_GET", querystring)
+        output = self.get_output_with_exceptions(response)
+        return output
+
+    def make_request(self, method, params):
+        response = {"text": ""}
+        try:
+            if method == "POST":
+                response = self.session.post(self.url, params=params, verify=self.verify_cert)
+            elif method == "GET":
+                response = self.session.post(self.url, params=params, verify=self.verify_cert)
+            elif method == "REQUEST_GET":
```

Comment:
What is a request_get?

Reply:
Can I get a comment here to let whoever's looking at this code know why that would show up?

Reply:
This is only some enum to check which method use. When `GET` we use: `session.get`, and when `REQUEST_GET` we use `request.get`. When we change this line: `self.session.post(self.url, params=params, verify=self.verify_cert)` we think it will be understandable enough. If in your opinion it will not be clear enough we will change code to be more transparent.